src/js/modules/controller/taskcontroller.js

```diff
@@ -0,0 +1,157 @@
+
+  static createTag = (tagName, textContent = null, className = null) => {
+    const tag = document.createElement(tagName);
+    tag.textContent = textContent;
+    tag.className = className;
+    return tag;
+  };
+
+  static createTaskRow = (task) => {
+    const taskRow = TaskController.createTag('li', null, 'task-item');
+    const taskForm = TaskController.createTag('form', null, 'task-form');
+    const descriptionElement = TaskController.createTag(
+      'input',
+      null,
+      'item description',
+    );
+    descriptionElement.type = 'text';
+    descriptionElement.name = 'description';
+    const indexElement = TaskController.createTag('div', null, 'item index');
+    const completedElement = TaskController.createTag('div', null, 'item completed');
+
+    const dragNdrop = TaskController.createTag(
+      'span',
+      null,
+      'fas fa-ellipsis-v vertical-3-dot',
+    );
+    const deleteIcon = TaskController.createTag(
+      'span',
+      null,
+      'fa fa-trash-o delete-icon',
+    );
+
+    const checkboxElement = TaskController.createTag('input', null, 'check-input');
+    checkboxElement.type = 'checkbox';
+    checkboxElement.name = 'completed';
+
+    checkboxElement.checked = task.completed;
+    /* checkboxElement.addEventListener("change", (event) => {
+      TaskController.resetForRemove();
+      const newTask = new Task(task.index, task.description, task.completed);
+      console.log('**********************')
+      console.log(newTask)
+      if(newTask){
+        newTask.setCompleted(checkboxElement.checked);
+        console.log(newTask)
+      }
+
+    }); */
+
+    descriptionElement.value = task.description;
+    indexElement.innerText = task.index;
+    completedElement.innerText = task.completed;
+
+    const rowItems = [
+      checkboxElement,
+      descriptionElement,
+      indexElement,
+      completedElement,
+    ];
+
+    for (let j = 0; j < rowItems.length; j += 1) {
+      taskForm.appendChild(rowItems[j]);
+    }
+
+    taskForm.addEventListener('click', () => {
+      TaskController.resetForRemove();
+      taskRow.classList.toggle('editing');
+      dragNdrop.classList.toggle('hidden');
+      deleteIcon.classList.toggle('show');
+    });
+
+    taskForm.addEventListener('submit', (event) => {
+      event.preventDefault();
+    });
+
+    descriptionElement.addEventListener('change', (event) => {
+      event.preventDefault();
+      const form = event.target.parentElement;
+      const index = parseInt(form.getElementsByTagName('div')[0].innerText, 10);
+      const description = form.description.value;
+      const completed = (form.getElementsByTagName('div')[1].innerText === 'true');
+      const task = new Task(index, description, completed);
+      task.save();
+    });
+
+    deleteIcon.addEventListener('click', () => {
+      TaskController.removeTask(TaskController.taskList, task);
+      // TaskController.removeTaskFromUI(event.target)
```

Comment:
Good job! but it's highly recommended to keep your repository clean. So please make sure to remove it.

Reply:
Fixed.